platform_secrets/api.py

```diff
@@ -106,57 +106,65 @@ def _secret_cluster_uri(self) -> str:
     def _get_org_secrets_uri(self, org_name: str) -> str:
         return f"{self._secret_cluster_uri}/{org_name}"
 
-    def _get_user_secrets_uri(self, user: User, org_name: Optional[str]) -> str:
+    def _get_secrets_uri(self, project_name: str, org_name: Optional[str]) -> str:
         if org_name is None:
             base = self._secret_cluster_uri
         else:
             base = self._get_org_secrets_uri(org_name)
-        return f"{base}/{user.name}"
+        return f"{base}/{project_name}"
 
-    def _get_user_secrets_read_perm(
-        self, user: User, org_name: Optional[str]
-    ) -> Permission:
-        return Permission(self._get_user_secrets_uri(user, org_name), "read")
+    def _get_secret_uri(self, secret: Secret) -> str:
+        base = self._get_secrets_uri(secret.project_name, secret.org_name)
+        if secret.owner == secret.project_name:
+            return f"{base}/{secret.key}"
+        return base
```

Comment:
I am wondering whether we could always unconditionally return `secret://cluster/org/project/key`? I guess same applied to `disks`.
From the `platform-auth` perspective, it will still be checking `secret://cluster/org/project` as part of the traversal logic.

Reply:
In such case we might want to enforce this on the auth side. otherwise users will still be able to "share" a URI, but that won't take effect and can cause some frustration.

Reply:
done